### PR TITLE
test: MAASENG-1276 update zoneselect tests

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done

- Refactored ZoneSelect tests to use testing library

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Pull this branch
- run `yarn && yarn test src/app/base/components/ZoneSelect`

## Fixes

Fixes: # .
[https://warthogs.atlassian.net/browse/MAASENG-1276](https://warthogs.atlassian.net/browse/MAASENG-1276)
## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.

### CODE REVIEW

File: src/app/base/components/ZoneSelect/ZoneSelect.test.tsx [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 import {
   rootState as rootStateFactory,
   zone as zoneFactory,
   zoneGenericActions as zoneGenericActionsFactory,
   zoneState as zoneStateFactory,
 } from "testing/factories";
+import { renderWithMockStore, screen } from "testing/utils";
 
-const mockStore = configureStore();
+const mockStore = configureStore<RootState>();
```
Comment (petermakowski) [inline src/app/base/components/ZoneSelect/ZoneSelect.test.tsx:15]: We're not directly interacting with the store in any of the test so you can remove this and pass the state to `renderWithMockStore` directly.

File: src/app/base/components/ZoneSelect/ZoneSelect.test.tsx [hunk 2]
```diff
@@ -25,29 +25,29 @@ describe("ZoneSelect", () => {
...
-          <ZoneSelect name="zone" />
-        </Formik>
-      </Provider>
-    );
 
+    renderWithMockStore(
+      <Formik initialValues={{ zone: "" }} onSubmit={jest.fn()}>
+        <ZoneSelect name="zone" />
+      </Formik>,
+      { store }
```
Comment (petermakowski) [inline src/app/base/components/ZoneSelect/ZoneSelect.test.tsx:49]:
```suggestion
```

File: src/app/base/components/ZoneSelect/ZoneSelect.test.tsx [hunk 3]
```diff
@@ -25,29 +25,29 @@ describe("ZoneSelect", () => {
...
 
-    expect(wrapper.find("select[name='zone']")).toMatchSnapshot();
+    expect(
+      screen.getByRole("combobox", { name: new RegExp("zone", "i") })
+    ).toMatchSnapshot();
   });
 
   it("dispatches action to fetch zones on load", () => {
     const state = rootStateFactory();
     const store = mockStore(state);
```
Comment (petermakowski) [inline src/app/base/components/ZoneSelect/ZoneSelect.test.tsx:43]:
```suggestion
```

File: src/app/base/components/ZoneSelect/ZoneSelect.test.tsx [hunk 4]
```diff
@@ -25,29 +25,29 @@ describe("ZoneSelect", () => {
...
+      <Formik initialValues={{ zone: "" }} onSubmit={jest.fn()}>
+        <ZoneSelect name="zone" />
+      </Formik>,
+      { state }
     );
 
-    expect(wrapper.find("select[name='zone']")).toMatchSnapshot();
+    expect(
+      screen.getByRole("combobox", { name: new RegExp("zone", "i") })
+    ).toMatchSnapshot();
```
Comment (petermakowski) [inline src/app/base/components/ZoneSelect/ZoneSelect.test.tsx:38]:
```suggestion
    expect(
      screen.getByRole("combobox", { name: /zone/i })
    ).toBeInTheDocument();
```

**Review by petermakowski — COMMENTED** [review]
Looks good, just a few comments below.

File: src/app/base/components/ZoneSelect/ZoneSelect.test.tsx (hunk 1 above)
Comment (Jay-Topher) [reply src/app/base/components/ZoneSelect/ZoneSelect.test.tsx:15]: How about line 52, where I have to test the store actions?

**Review by petermakowski — APPROVED** [review]
Thanks for making the changes. LGTM

**Pull request merged**